### PR TITLE
Fix l2c

### PR DESCRIPTION
**Pull request by pmiettinen**

### CODE REVIEW

File: peregrine/tracking.py [hunk 1]
```diff
@@ -207,8 +207,7 @@ def __init__(self, params):
     self.code_phase = 0.0
     self.carr_phase = 0.0
     self.samples_per_chip = int(round(self.sampling_freq / self.chipping_rate))
-    self.sample_index = params['samples']['sample_index']
-    self.sample_index += self.acq.sample_index
+    self.sample_index = self.acq.sample_index
```
Comment (adel-mamin) [inline peregrine/tracking.py]: Would it be possible to add 'sample_index' entry into 'params' dictionary and initialize it differently for L1C/A and L2C tracking channels:
L1C/A case: params['sample_index'] = params['samples']['sample_index']
L2C case: params['sample_index'] = 0 ?
Then this code would look like this:
self.sample_index = params['sample_index']

**Comment by adel-mamin** [comment]
In general, I think, it is better to separate the code formatting changes into a separate commit. It makes the review easier.

File: peregrine/defaults.py [hunk 2]
```diff
@@ -120,7 +120,7 @@
 
 l2c_loop_filter_params = {
     "loop_freq": 50,      # loop frequency [Hz]
-    "code_bw": 1.4,       # Code loop NBW
+    "code_bw": 1,       # Code loop NBW
```
Comment (adel-mamin) [inline peregrine/defaults.py]: This fix makes the 'code_bw' the same as it is in FW and it looks good to me.

**Comment by adel-mamin** [comment]
Looks good to me.

File: peregrine/tracking.py [hunk 3]
```diff
@@ -721,6 +721,7 @@ def __init__(self, params):
     params['code_freq_init'] = params['acq'].doppler * \
         gps_constants.l2c_chip_rate / gps_constants.l2
     params['chipping_rate'] = gps_constants.l2c_chip_rate
+    params['sample_index'] = 0
```
Comment (valeri-atamaniouk) [inline peregrine/tracking.py]: How to support the use case, when only L2C tracking is enabled, and offset is defined?

Comment (pmiettinen) [inline peregrine/tracking.py]: @adel-mamin ^

**Pull request merged**